File: Java/프로그래머스/0/181853. 뒤에서 5등까지/뒤에서 5등까지.java
import java.util.Arrays;

class Solution {
    private int NUM_RESULT = 5;
    
    public int[] solution(int[] num_list) {
        // 원본 배열의 변경 방지
        int[] n_list = num_list.clone();
        
        for (int i = 0; i < NUM_RESULT; i++) {
            int min_idx = i;
            for (int j = i; j < n_list.length; j++) {
                if (n_list[min_idx] > n_list[j]) {
                    min_idx = j;
                }
            }
            if (i != min_idx) {
                int temp = n_list[i];
                n_list[i] = n_list[min_idx];
                n_list[min_idx] = temp;
            }
        }
        int[] answer = Arrays.copyOfRange(n_list, 0, NUM_RESULT);
        
        return answer;
    }
}

File: Go/프로그래머스/0/181853. 뒤에서 5등까지/뒤에서 5등까지.go
var NUM_RESULT = 5

func solution(num_list []int) []int {
    // Go는 포인터를 전달하므로, 원본을 유지하기 위해 복사하여 사용함.
    n_list := make([]int, len(num_list))
    result := make([]int, NUM_RESULT)
    copy(n_list, num_list)
    
    // 선택 정렬을 NUM_RESULT(5)번만 수행
    for i := 0; i < NUM_RESULT; i++ {
        min_idx := i
        for j := i; j < len(n_list); j++ {
            if n_list[min_idx] > n_list[j] {
                min_idx = j
            }
        }
        if i != min_idx {
            n_list[min_idx], n_list[i] = n_list[i], n_list[min_idx]
        }
    }
    
    copy(result, n_list)
    
    return result
}